internal/go/skycfg/assert.go

```diff
@@ -0,0 +1,145 @@
+// Copyright 2018 The Skycfg Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package skycfg
+
+import (
+	"bytes"
+	"fmt"
+
+	"go.starlark.net/starlark"
+	"go.starlark.net/syntax"
+)
+
+// AssertModule contains assertion functions.
+// The *TestContext returned can be used to track assertion failures.
+// assert.* functions from this module will mutate the *TestContext.
+// After execution is complete, TestContext.Failures will be non-empty
+// if any of the assertions failed, and also contain details about the failures.
+func AssertModule() (*TestContext, starlark.Value) {
+	ctx := &TestContext{}
+	return ctx, &Module{
+		Name: "assert",
+		Attrs: starlark.StringDict{
+			"true": starlark.NewBuiltin("assert.true", ctx.AssertUnaryImpl),
```

Comment:
If `*TestContext` implements `starlark.Callable`, then you can have test code like this:

```python
def test_something(t):
  t.assert(True)
  t.assert.equal(1 + 2, 3)
```